src/ServiceWorker.js

```diff
@@ -27,34 +27,36 @@ const PRECACHE_URLS = [
 
 // The install handler takes care of precaching the resources we always need.
 self.addEventListener('install', event => {
-    // @ts-ignore
-    event.waitUntil(
-        caches.open(PRECACHE)
-            .then(cache => cache.addAll(PRECACHE_URLS))
-            // @ts-ignore
-            .then(self.skipWaiting()),
-    );
+    // @ts-ignore Property 'waitUntil' does not exist on type 'Event'.ts
+    event.waitUntil(async () => {
+        const cache = await caches.open(PRECACHE);
+        cache.addAll(PRECACHE_URLS);
```

Comment:
> See comments.
> Also instead of the `@ts-ignore`s try to cast events to `ExtendableEvent`s and `self` to `ServiceWorkerGlobalScope` if these types exist in typescript.

They don't :(